typed-protocols/src/Network/TypedProtocol/PingPong/Tests.hs

```diff
@@ -281,11 +285,21 @@ prop_channel (NonNegative n) = do
 
 prop_channel_IO :: NonNegative Int -> Property
 prop_channel_IO n =
-    ioProperty (prop_channel n)
-
-prop_channel_ST :: NonNegative Int -> Bool
-prop_channel_ST n = 
-    runSimOrThrow (prop_channel n)
+    ioProperty (prop_channel n nullTracer)
+
+prop_channel_ST :: NonNegative Int -> Property
+prop_channel_ST n =
+    ioProperty $ do
+        let (_msgs, bool) =
+                runSimOrThrow $ do
+                    c   <- newTVarM []
+                    res <- prop_channel n $ toListTracer c
+                    ms  <- atomically $ readTVar c
+                    return (ms, res)
+        -- forM_ (reverse _msgs) $ \case
+        --     Str m -> print m
+        --     Msg m -> print m
+        return $ property bool
```

Comment:
@coot With this change I wanted to demonstrate a use of `listTracer`. Should I revert these changes?

Reply:
Yes, I think you can remove it.

Reply:
I think for `SimM` directly we can use a more direct tracer, using the Sim's direct support of `Dynamic` in the trace output. It's like `MonadSay` but for any `Typeable`, and specific to the `SimM` monand, not with any class.

Reply:
@dcoutts Do you mean something like this?
```
dynTracer :: Typeable a => Tracer (SimM s) a
dynTracer = Tracer $ Op $ \a ->
                SimM $ \k -> Output (toDyn a) (k ())
```

Reply:
Yes.